src/libsyntax/parse/parser.rs

```diff
@@ -2160,9 +2160,11 @@ impl<'a> Parser<'a> {
                     suffix,
                 ) = self.token {
                     let suffix = suffix.and_then(|s| {
-                        let s = s.as_str().get();
-                        if ["f32", "f64"].contains(&s) {
```

Comment:
Hm, shouldn't this keep working with non-static lifetime?
Also, a typo "f64f64" below.

Reply:
No, the `LocalInternedString` is dropped in the first statement here.

Reply:
I assume this is going to ruin the `fn ident_str` API in https://github.com/rust-lang/rust/pull/58899 then.
Treating an attribute name as `&str` (including matching on it, calling `contains`) is a common operation, so returning `Option<LocalInternedString>` instead and `map`ping on it every time would be a serious usability hit.

Reply:
Yeah. `fn ident_str` won't work. We should really use symbols for attribute names instead of comparing strings all the time.

Reply:
True, the issue is just that a lot of existing code across rustc uses strings.

Reply:
Addressed in https://github.com/rust-lang/rust/pull/59256